### ISSUE TITLE
Partial credit memo tax calculation

### ISSUE THREAD

**Issue by lfritsche** [magento/magento2#8797]
### Preconditions

1. Magento 2.0.4
2. PHP 7

### Steps to reproduce

1. Order with more than one product | example: 
![image](https://cloud.githubusercontent.com/assets/15127451/23599754/2cdb0722-0243-11e7-917c-de8dd39b5652.png)
![image](https://cloud.githubusercontent.com/assets/15127451/23599773/4b7d63fa-0243-11e7-8973-df3b3efe18af.png)
2. Create partial credit memo manually in the backend:
![image](https://cloud.githubusercontent.com/assets/15127451/23599798/73c62540-0243-11e7-8b50-46c5dad92c2b.png)
![image](https://cloud.githubusercontent.com/assets/15127451/23599801/7dcba1aa-0243-11e7-84b2-912bd1c02d6f.png)
3. credit memo PDF output (style customized) :
![image](https://cloud.githubusercontent.com/assets/15127451/23599835/ad73e048-0243-11e7-91eb-0d1e39de5083.png)

### Expected result

1. Product Tax for the 5 refunded items = 2,93€ (7% reduced german tax on 44,87€)
2. Whole Tax amount for the order = 3,42€
3. PDF print displays the whole tax amount instead of the partial tax amount for the refunded items 
"reduzierte MwSt. 7%" displays 3,42€

### Actual result

1. "reduzierte MwSt. 7%" should display 2,93€
2. I'm also facing a similar issue using a third party extension, which can generate (partial) credit memos. If a partial credit memo is generated, the whole tax amount is added to the credit memo, instead of only the partial amount corresponding to the refunded items.

**Comment by vityakopin** [comment]
Thanks for reporting this issue. Please add more details to your description of the steps you followed when identifying this. Also provide tax configuration, price and shopping cart display settings and any information which will be helpful to reproduce this issue.

**Comment by lfritsche** [comment]
maybe this issue is related to https://github.com/magento/magento2/issues/9929, which could be resolved by commit https://github.com/magento/magento2/commit/ab0f7ef2996d63bac0e520f96140723cc9de92ef

**Comment by veloraven** [comment]
@lfritsche thank you for your report and your update.
Did mentioned commit resolve your problem?

**Comment by lfritsche** [comment]
@veloraven I have updated to 2.1.6 where the fix helped. For 2.0.4 I cannot test any more.

**Comment by andidhouse** [comment]
Wrong refund is also present in 2.1.7 - this really sucks.
![1](https://user-images.githubusercontent.com/16196483/30483640-e2a84ebc-9a27-11e7-86ae-4e0c60421274.png)
![2](https://user-images.githubusercontent.com/16196483/30483641-e2d64574-9a27-11e7-9e0a-125e919ae92e.png)

Refund without shipping costs should be clear 20,95 Euros but is 21,55 Euros.
Another bug reported month ago and not solved even in 2.1.9 - i am not shure why bugreuports get so permanently ignored here. 2.1.9 is a security update - really funny with over 2.000 bug reports open here...

**Comment by orlangur** [comment]
@andidhouse reporting problems in closed issues decreases probability of fix even more. Please report a new issue referring to this one.

**Comment by andidhouse** [comment]
thanks @orlangur i understand and it makes sense - sry.
But we won´t do it for this any more. Most of the bugs we reported (and there are 100% confirmed bugs like "active navigation class with varnish not working") so far are not solved and only internal tickets are created - some for over 1 year now. So our team does not see any sense on this any more as we do not need solutions in 1-2 years but 1-2 month - and magento proved it is not capable to deliver bugfixes in mid or short time (see the 2.1.9 release - a security release - really???).

Hope you understand.

**Comment by orlangur** [comment]
@andidhouse oki, no problem :) I don't find your attitude really collaborative but anyone definitely have a right to not report bugs. Let's see how the big issues cleanup worked in 6-9 months.

By the way, `2.1.10-preview` was created a long time ago and it was known `2.1.9` will be a security release.

**Comment by andidhouse** [comment]
@orlangur - we are totally disappointed here of the behavior of the magento team regarding the bug fixing - sry to be harsh sometimes here. Maybe if you see it from a agency / shop owner perspective you understand that more. We (our whole team) began to report bugs here far over 1 year now - others also did. 
And most of the bugs are still not fixed at all (indexing problems, wrong calculations, checkout problems etc.etc you name it!) also this are major bugs which cause major problems in the productivity of a commerce shop. 

I think the main problem is that a) magento2 has much more bugs then anyone expected and much more then the other comparable systems and b) nothing changes here if you look at new bugs introduced in 2.1.8 - each version introduced new ones.

I am not shure how you see this but i am shure this is a management fail. You have a great system but you do not get it to work properly (and i am not talking about small bugs - just take the big ones mentioned before).

So sry to sound harsh here sometimes but it is frustrating what is going on here since over 1.5 years now.

**Comment by korostii** [comment]
> it was known 2.1.9 will be a security release.

And yet includes a couple improvements not related to security, like logging the exceptions during checkout.

> By the way, 2.1.10-preview was created a long time ago

IMHO this just seems to confirm the fact that the current Magento's development cycle is way too slow and isn't capable of providing bugfixes in a timely manner.

**Comment by korostii** [comment]
@orlangur 
> reporting problems in closed issues decreases probability of fix even more. Please report a new issue referring to this one.

To be honest, I don't see how posting an extra report of the same bug is speeding things up.
I'd rather see the original issue reopened. If only we had someone with such power nearby :)

**Comment by orlangur** [comment]
@korostii easy: closed issues are not managed at all. The only situation when I think reopening would be useful is when some fix was mentioned in release notes but accidentally not merged into release.

Here you can see that issue is closed by author as upgrade did help (or issue disappeared due to some other reasons). Sometimes even if the issue contains similar scenario but not identical there could be 2+ different bugs.

In a bug report we need steps to reproduce straight from the horse's mouth which can be checked on the latest stable version and adjusted if another behavior occurs on another instance. Reopening an old issue would not bring any value here while opening a new one with relevant steps will definitely lead to bugfix.

UPD: by the way, 9929 is still open but seems to be fixed in `2.2.x` only yet. Instead of complaining how buggy `2.1.x` is one Community member shared the fix with another Community and it worked. Perfect example of collabotation 👍

**Comment by andidhouse** [comment]
I do not think this behavior makes any sense - you can see this now for over 1.5 years - you do not get to manage and fix bugs right in a valid time frame (you can see this here on many tickets - some small issues are solves on a topic and most of the time in a new version the same or a similar issue is introduced again.) - You can check this for nearly any bugs here - just type in for example "indexing", "reindex" - you see this is a permanent issue on all versions so far not solved clean in any version.

So your management should overthink the bug fixing here in principal. A bug should be reported only once here - and if it is fixed it should not be valid again in any new version - (otherwise the whole process makes no sense at all).

Why is it so complicated to have a workflow like this:

- You see that there are many bugs reported for example for the topic: "credit memo" - this proves there are errors in your coding - no discussion.
- Instead of fixing them one by one (and causing other new issues) your team should put all these bugs together and solve them together - then test all again and publish a clean working new version.

I think what we have seen so far here is way far away from productive in any way. 
Your team fixes one bug - which causes other new bugs etc.

In my opinion this is a bad management - and you see this here as there are over 2.000 bugreuports not solved - i think this speaks for itself.

Sry i do not agree with you that your process is good - and this is ongoing for over 1.5 years now.
